common/meta/grpc/src/grpc_client.rs

```diff
@@ -94,59 +114,91 @@ impl ItemManager for MetaChannelManager {
 
 pub struct MetaGrpcClient {
     conn_pool: Pool<MetaChannelManager>,
-    addr: String,
+    endpoints: RwLock<Vec<String>>,
+    #[allow(dead_code)]
+    cancel_tx: Sender<()>, // Implicit using when client dropped, notify backgroud task to stop.
     username: String,
     password: String,
-    token: Arc<RwLock<Option<Vec<u8>>>>,
+    token: RwLock<Option<Vec<u8>>>,
 }
 
 const AUTH_TOKEN_KEY: &str = "auth-token-bin";
 
 impl MetaGrpcClient {
     pub async fn try_new(
         conf: &MetaGrpcClientConf,
-    ) -> std::result::Result<MetaGrpcClient, Infallible> {
+    ) -> std::result::Result<Arc<MetaGrpcClient>, Infallible> {
         let mgr = MetaChannelManager {
             timeout: Some(Duration::from_secs(conf.client_timeout_in_second)),
             conf: conf.meta_service_config.tls_conf.clone(),
         };
-        Ok(Self {
+
+        let (tx, rx) = oneshot::channel();
+
+        let addr = conf.meta_service_config.address.to_string();
+        let endpoints = if !conf.meta_service_config.endpoints.is_empty() {
+            conf.meta_service_config.endpoints.clone()
+        } else {
+            vec![addr]
+        };
```

Comment:
Looks like it's time to remove `address`. Maybe in next PR.

Reply:
Hmm, if we remove `address` this PR, please let @ZhiHanZ know, the cloud platform will not work if we remove it.
I will add the databend query configuration to databend.rs soon.